design/src/aws_config.md

```diff
@@ -0,0 +1,161 @@
+# AWS Configuration RFC
+
+> Status: RFC. For an ordered list of proposed changes see: [Proposed changes](#changes-checklist).
+
+An AWS SDK loads configuration from multiple locations. Some of these locations can be loaded synchronously. Some are
+async. Others may actually use AWS services such as STS or SSO.
```

Comment:
For clarity purposes: does "configuration" primarily mean loading credentials and region details?

Reply:
yeah (and other stuff, but those are the two biggest and most complex)